### PR TITLE
[DO NOT MERGE BEFORE #366 & #367] Improve virtualized tree loading performances

### PR DESCRIPTION
**Pull request by leslielazzarino**
### Summary of changes

The tree item html & css part has been extracted to a new React component, which is managed by the virtualized list.

### Context and reason for change

Before all html & css items to be shown were generated at each rendering by VirtualizedTree and stored in an Array. That lead to complex non-modular code and a bunch of not-shown items being calculated in full, impacting performances. 

### How can the changes be tested

Basically a refactoring. The performance improvement wasn't meaningful on my laptop, even with a lot of expanded folders.

Note: Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

### CODE REVIEW

**Review by nicarl — APPROVED** [review]

File: src/Frontend/Components/VirtualisedTree/VirtualizedTreeItem.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      label={nodeId}
+    />
+  );
+}
+
+function isSelected(nodeId: string, selected: string): boolean {
+  return nodeId === selected;
+}
+
+export function isChildOfSelected(nodeId: string, selected: string): boolean {
```
Comment (benedikt-richter) [inline src/Frontend/Components/VirtualisedTree/VirtualizedTreeItem.tsx:127]: no export needed

File: src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx [hunk 2]
```diff
@@ -74,14 +53,16 @@ export function VirtualizedTree(
 ): ReactElement | null {
   const classes = useStyles();
 
-  const treeItems: Array<ReactElement> = props.resources
+  const treeItemProps: Array<VirtualizedTreeItemProps> = props.resources
```
Comment (benedikt-richter) [inline src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx:56]: The name `treeItemProps` suggest that this variable is an object. However, it's an array. Maybe `treeItemData` or something similar might be less missleading

**Review by benedikt-richter — CHANGES_REQUESTED** [review]
see comments

**Review by benedikt-richter — APPROVED** [review]
